ansible_base/jwt_consumer/common/auth.py

```diff
@@ -170,18 +171,23 @@ def validate_token(self, unencrypted_token, decryption_key):
         # Decrypt the token
         try:
             logger.info("Decrypting token")
-            validated_body = jwt.decode(
+            validated_body = self.decode_jwt_token(
                 unencrypted_token,
                 decryption_key,
-                audience="ansible-services",
                 options={"require": local_required_field},
-                issuer="ansible-issuer",
-                algorithms=["RS256"],
             )
         except jwt.exceptions.DecodeError as e:
             raise e  # This will be handled higher up
         except jwt.exceptions.ExpiredSignatureError:
-            self.log_and_raise(_("JWT has expired"))
+            expired_token = self.decode_jwt_token(
+                unencrypted_token,
+                decryption_key,
+                options={"require": local_required_field, "verify_exp": False},
+            )
+            expired_time = expired_token.get("exp")
+            now = datetime.now().timestamp()
+            time_diff = int(now - expired_time)
+            self.log_and_raise(_(f"JWT has expired {time_diff} seconds ago - check for clock skew"))
```

Comment:
fwiw from my perspective, its worth extra cost of doing this decoding a second time because if we are in this situation -- this is quite bad. 

Either
* clocks are out of sync
* component is so backlogged that tokens are arriving at it already expired
* some other bad/unsupported case of re-using old jwt tokens